clients/wallet/src/resubmissions.rs

```diff
@@ -0,0 +1,729 @@
+use std::sync::Arc;
+
+use crate::{
+	error::{
+		CacheError, CacheErrorKind, Error,
+		Error::{DecodeError, ResubmissionError},
+	},
+	StellarWallet, TransactionResponse,
+};
+use primitives::{
+	stellar::{Memo, Transaction, TransactionEnvelope, XdrCodec},
+	TransactionEnvelopeExt,
+};
+use std::time::Duration;
+use tokio::time::sleep;
+
+#[cfg(test)]
+use mocktopus::macros::mockable;
+
+#[cfg(not(test))]
+const RESUBMISSION_INTERVAL_IN_SECS: u64 = 1800;
+
+const MAX_LOOK_BACK_PAGES: u8 = 10;
+
+#[cfg_attr(test, mockable)]
+impl StellarWallet {
+	pub async fn resubmit_transactions_from_cache(&self) {
```

Comment:
Should we rename this function to indicate that this will actually start scheduling a periodic resubmission? Maybe we should also pass in the resubmission interval into this function call instead. Something like: `start_periodic_resubmission_of_transactions_from_cache(&self, interval_in_seconds: u64)`?

Reply:
renamed, and added the interval param